Client/Ecosia/UI/WhatsNew/DataProvider/WhatsNewLocalDataProvider.swift

```diff
@@ -71,43 +64,30 @@ final class WhatsNewLocalDataProvider: WhatsNewDataProvider {
         }
         return items
     }
-    
-    /// Evaluates if the stored app version needs to be updated to the current version.
-    ///
-    /// This function checks two conditions:
-    /// 1. If `fromVersion` is `nil`, indicating no version was previously stored.
-    /// 2. If `fromVersion` exists but is lower than `toVersion`, suggesting an app update.
-    ///
-    /// In either case, the stored app version is updated to the current version using the key `appVersionUpdateKey`.
-    ///
-    /// - Note: This function relies on the existence of `fromVersion` and `toVersion` properties in its scope and
-    /// the `Version` type's ability to update stored versions.
-    private func evaluateVersionNeedsUpdate() {
-        let isVersionNil = fromVersion == nil
-        let isVersionLowerThanCurrent = fromVersion != nil && fromVersion! < toVersion
-        let isVersionSameAsCurrent = fromVersion != nil && fromVersion! == toVersion
-                
-        if isVersionNil || (isVersionLowerThanCurrent && !isVersionSameAsCurrent) {
-            Version.updateFromCurrent(forKey: Self.appVersionUpdateKey)
-        }
-    }
-    
+        
     /// Private helper to fetch version range.
     ///
     /// - Returns: An array of `Version` between from and to, inclusive.
     func getVersionRange() -> [Version] {
-        
-        evaluateVersionNeedsUpdate()
-        
-        // Ensure fromVersion is available.
-        guard let fromVersion = self.fromVersion else { return [] }
-        
-        // If there's no update (i.e., the versions are the same), we shouldn't return any versions.
-        guard fromVersion != toVersion else { return [] }
-        
+                
+        // Ensure this is an upgrade scenario; otherwise, return an empty version range.
+        guard EcosiaInstallType.get() == .upgrade else { return [] }
+
+        // Ensure `fromVersion` is available; otherwise, return an empty version range.
+        guard let fromVersion else { return [] }
+
         // Gather all versions
         let allVersions = Array(whatsNewItems.keys).sorted()
-        
+                
+        // At this point in the logic, we are still in the upgrade scenario
+        // however, if the fromVersion and the toVersion will result being the same
+        // we will assume that we have upgraded to a version that didn't have this logic in place before
+        // There is no need to enforce the check for something ideally considered as a build update.
+        // As we currently checked the `.upgrade` scenario above.
+        if fromVersion == toVersion {
+            return allVersions.filter { $0 == toVersion } ?? []
```

Comment:
Does this also show items from previous versions? I guess there is no big problem if it doesn't since we only have one now, but it is something we can iterate on for the future.

Reply:
No, it doesn't. It will only show the items for the same version (if any) for now. Wanted to keep it super simple.
We hope that anybody will be soon on 9.0.0 so the scenario of an upgrade from 8.3.0 -> 9.x.x and/or 10+... will less likely be the case